pkg/assembler/assembler.go

```diff
@@ -17,43 +17,66 @@ package assembler
 
 type assembler struct{}
 
-// Identifiable implements the ability to retrieve a set of
-// attributes such that a graph query is able to identify a
-// GuacNode or GuacEdge uniquely (or as a GuacHyperNode).
-type Identifiable interface {
-	// Identifiers returns a map of fields and values which
-	// can be used to identify an object in the graph.
-	Identifiers() map[string]interface{}
-}
-
 // GuacNode represents a node in the GUAC graph
 type GuacNode interface {
-	Identifiable
-
 	// Type returns the type of node
 	Type() string
 
 	// Properties returns the list of properties of the node
 	Properties() map[string]interface{}
+
+	// Attributes returns the names of the properties of the node.
```

Comment:
These interfaces are likely going to have two types of uses, one is as a return to a query, where the returned object would act as an instance of a GuacNode. I think it's not super clear to me which of these functions relate to a schema vs relate to the actual instance of the object. 

i.e. should these calls return fields that may not be populated and Properties() return a map where they are set to the zero value? or should they return only if they are present. For the return value scenario i think it is pretty clear to me, but in the creation of an identifiable node for adding an edge for example, i still need to think how that would play out.

EDIT: now that i've seen the code below it seems like the former. That leaves the conversion to graph queries a little difficult now i think because its not certain which unique attributes exists (can't differentiate between missing or zero-value. e.g. int zero value is 0). Maybe `Get (name) error` instead of `Properties()`?.

Reply:
Oh, I'll need to think about missing fields. I picked `Properties` with this return type because this would have matched 1:1 to the type expected by the graphdb transactions.